src/modbus.c

```diff
@@ -387,6 +394,22 @@ static int mb_init_connection(mb_host_t *host) /* {{{ */
     return status;
   }
 
+#if defined(linux) && LIBMODBUS_VERSION_CHECK(2, 9, 4)
```

Comment:
Can you please update this line to match l1016:
`#if defined(linux) && !LEGACY_LIBMODBUS && LIBMODBUS_VERSION_CHECK(2, 9, 4)`

I think that will be more correct.

Reply:
This is inside a whole section of  `#ifdef !LEGACY_LIBMODBUS` because `mb_init_connection()` is declared once for `LEGACY_LIBMODBUS` and once for `!LEGACY_LIBMODBUS`. I added my code only to the function declared in the `!LEGACY_LIBMODBUS` case.

Reply:
Do you still like me do add this (redundant) `!LEGACY_LIBMODBUS`?

Reply:
No, thanks, I simply missed that upper 'if'.